lib/do_image/do_tag.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ *
+ * `triton image tag ...`
+ */
+
+var vasync = require('vasync');
+
+var common = require('../common');
+var errors = require('../errors');
+var mat = require('../metadataandtags');
+
+// ---- the command
+
+function do_tag(subcmd, opts, args, cb) {
+    if (opts.help) {
+        this.do_help('help', {}, [subcmd], cb);
+        return;
+    } else if (args.length < 1) {
+        cb(new errors.UsageError('incorrect number of args'));
+        return;
+    }
+
+    var log = this.top.log;
+    var tritonapi = this.top.tritonapi;
+
+    vasync.pipeline({arg: {cli: this.top}, funcs: [
+        common.cliSetupTritonApi,
+
+        function gatherTags(ctx, next) {
+            mat.tagsFromSetArgs(opts, args.slice(1), log, function (err, tags) {
+                if (err) {
+                    next(err);
+                    return;
+                }
+                log.trace({tags: tags || '<none>'},
+                    'tags loaded from opts and args');
+                ctx.tags = tags;
+                next();
+            });
+        },
+        function updateImageTags(ctx, next) {
+            log.trace({dryRun: opts.dry_run, tags: ctx.tags},
+                'image tag');
+
+            if (opts.dry_run) {
+                next();
+                return;
+            }
+
+            tritonapi.updateImage({
+                image: args[0],
+                fields: {
+                    tags: ctx.tags
+                }
+            }, function (err, img) {
+                if (err) {
```

Comment:
I wonder if there is any value in:
    `triton image tag $IMG name=value`
over
    `triton image update $IMG tags='{"name": "value"}'`

Reply:
Simplicity? I mean, I find the first command easier to run from a shell and not having to deal with JSON is a plus, IMO. I find the update subcommand sightly more complex and targeting changes of other fields more than tags or acls

Reply:
Okay - that is true.